corehq/apps/reports/views.py

```diff
@@ -544,10 +544,11 @@ def user(self):
 
 @login_and_domain_required
 @datespan_default
+@require_POST
```

Comment:
I think this should likely be first? The way I'm reading this, If the user sends a GET request with an invalid login, they will receive a login error, rather than an invalid method error. I think we want to indicate this endpoint doesn't even exist as a GET request, so checking that it's a POST request should be our first check

Reply:
This looks inconsistent enough that I think we should just do it in the most correct way -- there doesn't seem to be a real convention. But from what I can tell, invalid request methods should immediately terminate requests before any additional processing happens.

Reply:
Sounds reasonable to me. I'll make a PR to address this.